lnbits/wallets/breez.py

```diff
@@ -0,0 +1,177 @@
+    ):
+        def on_event(self, event):
+            logger.debug(event)
+            breez_event_queue.put_nowait(event)
+
+    class BreezSdkWallet(Wallet):  # type: ignore[no-redef]
+        def __init__(self):
+            if not settings.breez_greenlight_seed:
+                raise ValueError(
+                    "cannot initialize BreezSdkWallet: missing breez_greenlight_seed"
+                )
+            if not settings.breez_api_key:
+                raise ValueError(
+                    "cannot initialize BreezSdkWallet: missing breez_api_key"
+                )
+            if not settings.breez_greenlight_invite_code:
+                raise ValueError(
+                    "cannot initialize BreezSdkWallet: "
+                    "missing breez_greenlight_invite_code"
+                )
+
+            self.config = breez_sdk.default_config(
+                breez_sdk.EnvironmentType.PRODUCTION,
+                settings.breez_api_key,
+                breez_sdk.NodeConfig.GREENLIGHT(
+                    config=breez_sdk.GreenlightNodeConfig(
+                        partner_credentials=None,
+                        invite_code=settings.breez_greenlight_invite_code,
+                    )
+                ),
+            )
+
+            breez_sdk_working_dir = Path(settings.lnbits_data_folder, "breez-sdk")
+            breez_sdk_working_dir.mkdir(parents=True, exist_ok=True)
+            self.config.working_dir = breez_sdk_working_dir.absolute().as_posix()
+
+            seed = breez_sdk.mnemonic_to_seed(settings.breez_greenlight_seed)
+            self.sdk_services = breez_sdk.connect(self.config, seed, SDKListener())
+
+        async def cleanup(self):
+            self.sdk_services.disconnect()
+
+        async def status(self) -> StatusResponse:
+            try:
+                node_info: breez_sdk.NodeState = self.sdk_services.node_info()
+            except Exception as exc:
+                return StatusResponse(f"Failed to connect to breez, got: '{exc}...'", 0)
+
+            return StatusResponse(None, int(node_info.max_payable_msat))
+
+        async def create_invoice(
+            self,
+            amount: int,
+            memo: Optional[str] = None,
+            description_hash: Optional[bytes] = None,
+            unhashed_description: Optional[bytes] = None,
+            **kwargs,
+        ) -> InvoiceResponse:
+            # if description_hash or unhashed_description:
+            #     raise Unsupported("description_hash and unhashed_description")
+
+            breez_invoice: breez_sdk.ReceivePaymentResponse = (
+                self.sdk_services.receive_payment(
+                    breez_sdk.ReceivePaymentRequest(
+                        amount * 1000,  # breez uses msat
+                        memo,
+                        preimage=kwargs.get("preimage"),
+                        opening_fee_params=None,
+                    )
+                )
+            )
```

Comment:
I believe here you could add setting the description hash the same way as corelightning.
```suggestion
            if description_hash and not unhashed_description:
                raise Unsupported(
                    "'description_hash' unsupported by Greenlight, provide"
                    " 'unhashed_description'"
                )
            breez_invoice: breez_sdk.ReceivePaymentResponse = (
                self.sdk_services.receive_payment(
                    breez_sdk.ReceivePaymentRequest(
                        amount * 1000,  # breez uses msat
                        unhashed_description.decode() if unhashed_description else memo,
                        preimage=kwargs.get("preimage"),
                        opening_fee_params=None,
                        use_description_hash=True if unhashed_description else None,
                    )
                )
            )
```